### PR TITLE
feat(atomic): numeric facet

### PR DESCRIPTION
**Pull request by nrawji**
KIT-454
https://coveord.atlassian.net/browse/KIT-454

### CODE REVIEW

**Review by samisayegh — COMMENTED** [review]
A general comment about the PR is it is too long to provide an effective review. The sum of the red & green diffs should ideally not exceed 400 lines. This PR is more than 3x the soft limit. 

It's possible to stay within by focusing on a single concern. For a PR dedicated to the numeric facet, there should be no changes related to the search-box, or to the facet search which the numeric facet does not have.

It is okay for this one but please do not make it the norm. Smaller PRs allow a team to give better feedback, but also make it easier for the reviewee to make changes when needed. They are a much better way to collaborate, for everyone involved in the process.

File: packages/atomic/src/components/facets/atomic-numeric-facet-range/atomic-numeric-facet-range.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import {Component, Element, Prop} from '@stencil/core';
+
+/**
+ * Component that defines a numeric facet range.
+ * Has to be used inside an `atomic-numeric-facet` component.
+ */
+@Component({
+  tag: 'atomic-numeric-facet-range',
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply packages/atomic/src/components/facets/atomic-numeric-facet-range/atomic-numeric-facet-range.tsx:8]: I'd prefer `atomic-numeric-facet`

File: packages/atomic/src/components/facets/base-facet/base-facet.tsx [hunk 2]
```diff
@@ -53,7 +58,7 @@ export const BaseFacet: FunctionalComponent<BaseFacetProps> = (
     <div class="facet" part="facet">
       <button
         class={
-          'facet-button border-solid bg-transparent  px-4 h-9 outline-none focus:outline-none lg:hidden cursor-pointer ' +
+          'facet-button border-solid bg-transparent px-4 h-9 outline-none focus:outline-none lg:hidden cursor-pointer ' +
```
Comment (olamothe) [inline packages/atomic/src/components/facets/base-facet/base-facet.tsx:61]: Is `cursor-pointer` needed if it's already a button ?

**Review by olamothe — APPROVED** [review]

File: packages/atomic/src/components/facets/atomic-numeric-facet/atomic-numeric-facet.tsx [hunk 3]
```diff
@@ -13,37 +13,70 @@ import {
...
+   * Whether or not the index should automatically generate options for the facet
+   */
+  @Prop() public generateAutomaticRanges = true;
+
+  public buildOptions() {
+    const options = Array.from(
+      this.host.querySelectorAll('atomic-numeric-facet-range')
+    );
+    return options.map(({start, end, endInclusive}) =>
+      buildNumericRange({start, end, endInclusive})
```
[the remark this answers is not in the document]
Comment (nrawji) [reply packages/atomic/src/components/facets/atomic-numeric-facet/atomic-numeric-facet.tsx:70]: I tried this but the problem is that only properties get exposed inside the `component.d.ts` file so Typescript doesn't know that a range attribute exists for `atomic-numeric-range`.

**Review by ThibodeauJF — APPROVED** [review]

**Pull request merged**